docs/source/API/core/execution_spaces.rst

```diff
@@ -210,16 +210,9 @@ and an instance of that type ``ex``, Kokkos guarantees the following expressions
 
 .. code-block:: cpp
 
-    ex.in_parallel();
+    ex.fence(str);
 
-*Returns:* a value convertible to ``bool`` indicating whether or not the caller is executing as part of a Kokkos parallel pattern.
-*Note:* as currently implemented, there is no guarantee that ``true`` means the caller is necessarily executing as part of a pattern on the particular instance ``ex``; just *some* instance of ``Ex``. This may be strengthened in the future.
-
-.. code-block:: cpp
-
-    ex.fence();
-
-*Effects:* Upon return, all parallel patterns executed on the instance ``ex`` are guaranteed to have completed, and their effects are guaranteed visible to the calling thread.
+*Effects:* Upon return, all parallel patterns executed on the instance ``ex`` are guaranteed to have completed, and their effects are guaranteed visible to the calling thread. The optiopnal ``str`` argument allows customizing the event reported to Kokkos Tools.
```

Comment:
Should we be saying something about deep copies?

Reply:
```suggestion
*Effects:* Upon return, all parallel patterns and deep_copy calls executed on the instance ``ex`` are guaranteed to have completed, and their effects are guaranteed visible to the calling thread. The optiopnal ``str`` argument allows customizing the event reported to Kokkos Tools.
```

Reply:
Sure.